server/server/user.py

```diff
@@ -0,0 +1,128 @@
+    """Class which handles and setups the incoming client connection"""
+
+    @classmethod
+    async def create(
+        cls,
+        session_id: str,
+        websocket: WebSocket,
+        db: managers.DbManager,
+        connections: managers.ConnectionManager,
+    ):
+        """Asynchronous __init__ of User class"""
+        self = User()
+        self.session_id = session_id
+        self.connections = connections
+        self.websocket = websocket
+        await self.websocket.accept()
+        self.logged_in = False
+        self.db = db
+        self.close = False
+        yield self
+        self.user_id = await self.wait_for_auth()
+        await self.handle_user(self.user_id)
+
+    @websocket_connection
+    async def wait_for_auth(self):
+        """Authenticates(login or register) incoming connections from a user"""
+        retry = True
+        while retry or not self.close:
+            request = await self.websocket.receive_json()
+            user_data = {}
+            if request["type"] == "user.login":
+                users = self.db.get_user()
+                for i in users:
+                    if users[i]["username"] == request["username"]:
+                        if users[i]["password"] == request["password"]:
+                            user_data["user_id"] = users[i]["user_id"]
+                            user_data["username"] = users[i]["username"]
+                            user_data["rooms"] = self.db.get_user_rooms(
+                                user_data["user_id"]
+                            )
+                            await self.websocket.send_json(
+                                {"type": "user.login.success", "data": user_data}
+                            )
+                            self.logged_in = True
+                            retry = False
+                            return user_data["user_id"]
+                else:
+                    await self.websocket.send_json(
+                        {
+                            "type": "user.login.rejected",
+                            "data": None,
+                            "message": "Username or Password is wrong",
+                        }
+                    )
+                    continue
+            elif request["type"] == "user.register":
+                user_id = self.db.create_user(request["username"], request["password"])
+                user_data["user_id"] = user_id
+                await self.websocket.send_json(
+                    {
+                        "type": "user.register.success",
+                        "data": user_data,
+                        "message": "registered succesfully",
+                    }
+                )
+
+    @websocket_connection
+    async def handle_user(self, user_id):
+        """Handles requests from a logged in user"""
+        while True and not self.close:
+            request = await self.websocket.receive_json()
+            if request["type"] == "msg.send":
```

Comment:
add a option to send `json` as it is 
dont check for type just send to the other user

Reply:
we need to store the message in database before sending